src/raw/allocated.rs

```diff
@@ -5,71 +5,78 @@ use core::mem::{transmute, MaybeUninit};
 use core::ops::Range;
 use core::panic::{RefUnwindSafe, UnwindSafe};
 
+// TODO remove once provenance API stabilized
+use sptr::Strict;
+
 use crate::alloc::vec::Vec;
+use crate::backend::rc::Inner;
 use crate::backend::{rc, Backend};
 
 const MASK: usize = super::MASK as usize;
 const TAG: usize = super::TAG_ALLOCATED as usize;
 
-struct TaggedRaw<B: Backend>(*mut (), PhantomData<rc::Raw<Vec<u8>, B>>);
+struct TaggedRawRc<B: Backend>(usize, PhantomData<rc::Raw<Vec<u8>, B>>);
```

Comment:
If this is a "pointer or integer" value, it should use `*mut ()` not `usize`, as explained in https://github.com/rust-lang/rust/pull/130350.

Reply:
At some point, I need to see it as a `NonZero<u8> || [u8; _ ]` to maintain the same niche for all the representation. If I transmute `TaggedRawRc` to the this pivot representation I lose the provenance. But if I first expose the pointer, it "works" for now at least.